src/services/item/plugins/recycled/repository.ts

```diff
@@ -33,29 +34,41 @@ export class RecycledItemDataRepository extends MutableRepository<RecycledItemDa
     return recycled;
   }
 
-  async getOwn(account: Account, pagination: Pagination): Promise<Paginated<RecycledItemData>> {
+  async getOwnRecycledItems(account: Account, pagination: Pagination): Promise<Paginated<Item>> {
     const { page, pageSize } = pagination;
     const limit = Math.min(pageSize, ITEMS_PAGE_SIZE_MAX);
     const skip = (page - 1) * limit;
-    // get only with admin membership
-    const query = this.repository
-      .createQueryBuilder('recycledItem')
+
+    const query = this.manager
+      // start with smaller table that can have the most contraints: membership with admin and accountId
+      .getRepository(ItemMembership)
+      .createQueryBuilder('im')
+      // we want to join on recycled item
       .withDeleted()
-      .leftJoinAndSelect('recycledItem.creator', 'creator')
-      .leftJoinAndSelect('recycledItem.item', 'item')
-      .leftJoinAndSelect('item.creator', 'itemMember')
-      .innerJoin(
-        'item_membership',
-        'im',
-        `im.item_path @> item.path 
-        AND im.permission = :permission 
-        AND im.account_id = :accountId`,
-        { permission: PermissionLevel.Admin, accountId: account.id },
+      .innerJoinAndSelect(
+        'im.item',
+        'item',
+        // reduce size by getting only recycled items
+        `item.path <@ im.item_path and item.deleted_at is not null`,
       )
-      .orderBy('recycledItem.created_at', 'DESC');
+      // get top most recycled item
+      .innerJoin(RecycledItemData, 'rid', 'item.path = rid.item_path')
+      // return item's creator
+      .leftJoinAndSelect('item.creator', 'member')
+      // item membership constraints
+      .where(`im.account_id = :accountId`, {
+        accountId: account.id,
+      })
+      .andWhere(` im.permission = :permission`, {
+        permission: PermissionLevel.Admin,
+      })
+      // show most recently deleted items first
+      .orderBy('item.deleted_at', 'DESC')
+      .offset(skip)
+      .limit(limit);
 
-    const [data, totalCount] = await query.offset(skip).limit(limit).getManyAndCount();
-    return { data, totalCount, pagination };
+    const [data, totalCount] = await query.getManyAndCount();
```

Comment:
I think we will have issues with getting the count with this query too. It does not really make sense to optimize a query with pagination if we perform a query that materializes all rows just to get their count. 

If we get the count just for the frontend to show the last page, I think it is not worth it. I guess having an infinite query would be enough in the frontend. What do you think ?